### PR TITLE
Recipe-App 

### PR DESCRIPTION
**Pull request by Santiago220991**
# In this branch we:

- Use Ruby gems as a software packages system.
- Install Ruby on Rails framework.
- Understand Rails RESTful design and router.
- Use controllers to handle requests and render empty views.
- Understand Rails naming conventions.
- Use params from browser requests in a safe way.
- Use preprocessed HTML file with embedded Ruby code.
- Use layouts and templates for shared content.
- Use database migration files to maintain database schema.
- Use validations for models.
- Secure app from n+1 problems.
- Understand what ORM is.
- Write SQL queries with ActiveRecord.
- Set up associations between models.
- Build a web app that requires the user to log in.
- Describe the difference between authorization and authentication.
- Use Devise gem for authentication.
- Understand how sessions and cookies can support authentication.
- Limit access to web app resources based on authorization rules.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,51 @@
-# Recipe-App
\ No newline at end of file
+![](https://img.shields.io/badge/Microverse-blueviolet)
+
+# Item catalog
+
+> The Recipe app keeps track of all your recipes, ingredients, and inventory. It will allow you to save ingredients, keep track of what you have, create recipes, and generate a shopping list based on what you have and what you are missing from a recipe. Also, since sharing recipes is an important part of cooking the app should allow you to make them public so anyone can access them.
+
+<img src="./recipe_erd_2_members.png" alt="C=UML class diagram for catalog of things" />
```
Comment (Rustamxon7) [inline README.md:7]: That's a great project! I like it. You both did a great job here!

Great documentation 🏆
No N+1 problem 🏆
No Linter errors 🏆

However great project comes with great responsibilities 😜. I'm just kidding. Or, I think I wanted to say, In big cities, there might be small problems as well. Ok, let's begin...

File: spec/integration/general_shopping_list_spec.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+RSpec.describe 'General Shopping List', type: :feature do
+  describe 'index page' do
+    before(:each) do
+      User.destroy_all
+      Food.destroy_all
+      Recipe.destroy_all
+      RecipeFood.destroy_all
+      user = User.create(name: 'user', email: 'user@gmail.com', password: 123_456)
+      food = Food.create(name: 'meat', measurement_unit: 'lib', price: 1, quantity: 3, user:)
```
Comment (Rustamxon7) [inline spec/integration/general_shopping_list_spec.rb:11]: - [ ] Are you sure `user:` should stay like that? I'm sure you missed something or you wanted to put it but were not able to do it before I review it 😊.  Kindly complete the initialization.

File: spec/integration/general_shopping_list_spec.rb [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+      recipe = Recipe.create(name: 'bulgogi', preparation_time: 2, cooking_time: 1, description: 'good', public: true,
+                             user:)
```
Comment (Rustamxon7) [inline spec/integration/general_shopping_list_spec.rb:13]: - [ ] Oh, not again the same thing 🤦🏻‍♂️. Make sure you completed it by adding an argument to the value.

File: spec/integration/general_shopping_list_spec.rb [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+      RecipeFood.create(quantity: 5, recipe:, food:)
```
Comment (Rustamxon7) [inline spec/integration/general_shopping_list_spec.rb:14]: - [ ] Last but not least this is the final thing that you needed to fix.

**Review by rustam-tolipov — CHANGES_REQUESTED** [review]
Hello Team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
_Your project is almost complete!_
_No Linters errors_


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Santiago220991** [comment]
Hi @Rustamxon7. We are going to fix it right now. Thanks. :+1:

**Review by EricMbouwe — APPROVED** [review]
Hi @Santiago220991, 

Your project is complete! ⚡🔥  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 🎉🔥
- Requested changes ✔

## Optional suggestions ☂

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**